libr/anal/p/anal_tricore.c

```diff
@@ -0,0 +1,140 @@
+#include <string.h>
+#include <r_types.h>
+#include <r_lib.h>
+#include <r_asm.h>
+#include <r_anal.h>
+#include "../../asm/arch/tricore/gnu/tricore-opc.c"
+
+static bool set_reg_profile(RAnal *anal) {
+
+    const char *p = NULL;
+	switch (anal->bits) {
+	case 32: p =
+		"=PC	pc\n"
+		"=SP	a10\n"
+		"gpr	a0	.32	0	0\n"
+		"gpr	a1	.32	32	0\n"
+		"gpr	a2	.32	64	0\n"
+		"gpr	a3	.32	96	0\n"
+		"gpr	a4	.32	128	0\n"
+		"gpr	a5	.32	160	0\n"
+		"gpr	a6	.32	192	0\n"
+		"gpr	a7	.32	224	0\n"
+		"gpr	a8	.32	256	0\n"
+		"gpr	a9	.32	288	0\n"
+		"gpr	a10	.32	320	0\n"
+		"gpr	a11	.32	352	0\n"
+		"gpr	a12	.32	384	0\n"
+		"gpr	a13	.32	416	0\n"
+		"gpr	a14	.32	448	0\n"
+		"gpr	a15	.32	480	0\n"
+		"gpr	d0	.32	512	0\n"
+		"gpr	d1	.32	544	0\n"
+		"gpr	d2	.32	576	0\n"
+		"gpr	d3	.32	608	0\n"
+		"gpr	d4	.32	640	0\n"
+		"gpr	d5	.32	672	0\n"
+		"gpr	d6	.32	704	0\n"
+		"gpr	d7	.32	736	0\n"
+		"gpr	d8	.32	768	0\n"
+		"gpr	d9	.32	800	0\n"
+		"gpr	d10	.32	832	0\n"
+		"gpr	d11	.32	864	0\n"
+		"gpr	d12	.32	896	0\n"
+		"gpr	d13	.32	928	0\n"
+		"gpr	d14	.32	960	0\n"
+		"gpr	d15	.32	992	0\n"
+		"gpr	PSW	.32	1024	0\n"
+		"gpr	PCXI	.32	1056	0\n"
+		"gpr	FCX	.32	1088	0\n"
+		"gpr	LCX	.32	1120	0\n"
+		"gpr	ISP	.32	1152	0\n"
+		"gpr	ICR	.32	1184	0\n"
+		"gpr	PIPN	.32	1216	0\n"
+		"gpr	BIV	.32	1248	0\n"
+		"gpr	BTV	.32	1280	0\n"
+		"gpr	pc	.32	1312	0\n";
+	        break;
+	case 64: p =
+		"=PC	pc\n"
+		"=SP	a10\n"
+		"gpr	e0	.64	0	0\n"
+		"gpr	d0	.32	0	0\n"
+		"gpr	d1	.32	32	0\n"
+		"gpr	e2	.64	64	0\n"
+		"gpr	d2	.32	64	0\n"
+		"gpr	d3	.32	96	0\n"
+		"gpr	e4	.64	128	0\n"
+		"gpr	d4	.32	128	0\n"
+		"gpr	d5	.32	160	0\n"
+		"gpr	e6	.64	192	0\n"
+		"gpr	d6	.32	192	0\n"
+		"gpr	d7	.32	224	0\n"
+		"gpr	e8	.64	256	0\n"
+		"gpr	d8	.32	256	0\n"
+		"gpr	d9	.32	288	0\n"
+		"gpr	e10	.64	320	0\n"
+		"gpr	d10	.32	320	0\n"
+		"gpr	d11	.32	352	0\n"
+		"gpr	e12	.64	384	0\n"
+		"gpr	d12	.32	384	0\n"
+		"gpr	d13	.32	416	0\n"
+		"gpr	e14	.64	448	0\n"
+		"gpr	d14	.32	448	0\n"
+		"gpr	d15	.32	480	0\n"
+		"gpr	p0	.64	512	0\n"
+		"gpr	a0	.32	512	0\n"
+		"gpr	a1	.32	544	0\n"
+		"gpr	p2	.64	576	0\n"
+		"gpr	a2	.32	576	0\n"
+		"gpr	a3	.32	608	0\n"
+		"gpr	p4	.64	640	0\n"
+		"gpr	a4	.32	640	0\n"
+		"gpr	a5	.32	672	0\n"
+		"gpr	p6	.64	704	0\n"
+		"gpr	a6	.32	192	0\n"
+		"gpr	a7	.32	224	0\n"
+		"gpr	p8	.64	768	0\n"
+		"gpr	a8	.32	768	0\n"
+		"gpr	a9	.32	800	0\n"
+		"gpr	p10	.64	832	0\n"
+		"gpr	a10	.32	832	0\n"
+		"gpr	a11	.32	864	0\n"
+		"gpr	p12	.64	896	0\n"
+		"gpr	a12	.32	896	0\n"
+		"gpr	a13	.32	928	0\n"
+		"gpr	p14	.64 	960	0\n"
+		"gpr	a14	.32	960	0\n"
+		"gpr	a15	.32	992	0\n"
+		"gpr	PSW	.32	1024	0\n"
+		"gpr	PCXI 	.32	1056	0\n"
+		"gpr	FCX	.32	1088	0\n"
+		"gpr	LCX	.32	1120	0\n"
+		"gpr	ISP	.32	1152	0\n"
+		"gpr	ICR	.32	1184	0\n"
+		"gpr	PIPN 	.32	1216	0\n"
+		"gpr	BIV	.32	1248	0\n"
+		"gpr	BTV	.32	1280	0\n"
+		"gpr	pc	.32	1312	0\n";
+		break;
+	}
+	return r_reg_set_profile_string (anal->reg, p);
+}
+
+struct r_anal_plugin_t r_anal_plugin_tricore = {
```

Comment:
Use camel case syntax

Reply:
Didn't get you? 
I have copied these format from other plugins.

Reply:
use RAnalPlugin instead of 'struct r_anal_plugin_t'